runners/direct-java/src/main/java/org/apache/beam/runners/direct/DirectMetrics.java

```diff
@@ -76,6 +64,8 @@
   private static class DirectMetric<UpdateT, ResultT> {
     private final MetricAggregation<UpdateT, ResultT> aggregation;
 
+    private final ExecutorService executorService;
```

Comment:
Worth noting that the lifecycle of the ExecutorService is managed outside of this class; Alternatively you could pass the `submit` method as a `Consumer<Runnable>` or use that type for the field, to ensure that future changes are clear that this can't be closed from this class.

Reply:
Fair enough, will move to Executor